crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -203,6 +204,36 @@ impl ChunkCast for Utf8Chunked {
                     polars_bail!(ComputeError: "expected 'precision' or 'scale' when casting to Decimal")
                 },
             },
+            #[cfg(feature = "dtype-date")]
+            DataType::Date => {
+                let result = cast_chunks(&self.chunks, data_type, true)?;
+                let out = Series::try_from((self.name(), result))?;
+                Ok(out)
+            },
+            #[cfg(feature = "dtype-datetime")]
+            DataType::Datetime(_tu, tz) => {
+                let out = match tz {
+                    #[cfg(feature = "timezones")]
+                    Some(tz) => {
+                        validate_time_zone(tz)?;
+                        let result = cast_chunks(
+                            &self.chunks,
+                            &Datetime(TimeUnit::Nanoseconds, Some(tz.clone())),
```

Comment:
Why nanosecond instead of preserving the time unit?

Reply:
The cast function on arrow2 does not support other time units, I open an issue for that [here](https://github.com/jorgecarleitao/arrow2/issues/1541). 

Here is the arrow2 code: https://github.com/jorgecarleitao/arrow2/blob/c1446fb17229f1e5e703a8623b0946c6c8f5c3df/src/compute/cast/mod.rs#L597

Reply:
Thanks for explaining! I thought the idea was to go down the same path that `.straight.strptime` would go down?

Reply:
We should then first implement that upstream. @brayanjuls can you add the proper cast support in arrow2?

Reply:
@MarcoGorelli I thought about that but after inspecting the code I saw the functions supporting str.strptime are out of scope of the ones used when executing col("x").cast(pl.Date), the former are implemented in the StringFunction enum in the plan crates and the later are implemented in the Utf8Chunked struct in the core crate. It seems to me that I can’t access the plan crate modules from the core crate. I can try to duplicate the code in the core crate but it doesn’t seems right to me. If you have in mind a better way of implementing this please let me know, I am new to the code and to rust but happy to learn. 

@ritchie46 Yes, I am going to send a PR this next week, and modify this PR accordingly.

Reply:
@ritchie46 I provided a PR on arrow2, I need some help there but I think it is almost done, if you can review it would be great. https://github.com/jorgecarleitao/arrow2/pull/1560

Reply:
@brayanjuls should this be integrated directly into Polars since arrow2 got forked into nano-arrow? https://github.com/pola-rs/polars/pull/11179